zap/src/main/java/org/parosproxy/paros/network/HttpSender.java

```diff
@@ -259,6 +262,24 @@ private void checkState() {
         }
     }
 
+    private void updateCookieState() {
+        if (!useCookies) {
+            // discard cookies, but keep credentials
+            HttpState state = new HttpState();
+            HttpState proxyState = new HttpState();
+
+            state.setCredentials(AuthScope.ANY, client.getState().getCredentials(AuthScope.ANY));
```

Comment:
This should not be needed, we don't set credentials (at least not to one of these states directly).